2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,49 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    local cmdOutput
+    cmdOutput="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
+
+    # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
+    if addr="$(echo "$cmdOutput" | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+$')"; then
+      printf "$addr"
+      exit 0
+    fi
+
+    echo >&2 "Cannot get address of primary from $mongo_node node: $cmdOutput"
+    exit 1
+  done && return
+
+  echo >&2 "Cannot get address of primary node: after checking all nodes we don't have the address"
+
+  # return empty string
+  echo
```

Comment:
>exit code of mongo_primary_member_addr should be according to success or not

@rhcarvalho @legionus I not agree with that I don't see benefit of this. In caller's method we always analyze the output of the method and adding exit code will means that I also add code for ignoring exit code.